`lightning.tensor` mps docs (#756)
PennyLaneAI/pennylane-lightning: README.rst

* ``lightning.qubit``: is a fast state-vector simulator written in C++.
* ``lightning.gpu``: is a state-vector simulator based on the `NVIDIA cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_. It notably implements a distributed state-vector simulator based on MPI.
* ``lightning.kokkos``: is a state-vector simulator written with `Kokkos <https://kokkos.github.io/kokkos-core-wiki/index.html>`_. It can exploit the inherent parallelism of modern processing units supporting the `OpenMP <https://www.openmp.org/>`_, `CUDA <https://developer.nvidia.com/cuda-toolkit>`_ or `HIP <https://docs.amd.com/projects/HIP/en/docs-5.3.0/index.html>`_ programming models.

> * ``lightning.tensor``: is a tensor network simulator based on the `NVIDIA cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_. This device is designed to simulate large-scale quantum circuits using tensor networks. For small circuits, state-vector simulator plugins may be more suitable. The supported method is Matrix Product State (MPS). This device does not currently support finite shots. At present, the supported measurement types are expectation values and the supported differentiation method are ``parameter-shift`` or ``finite-diff``. Note that only 1,2-wires gates and gates that can be decomposed into 1,2-wires gates are supported.

Reviewer: ```suggestion
* ``lightning.tensor``: is a tensor network simulator based on the `NVIDIA cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_. This device is designed to simulate large-scale quantum circuits using tensor networks. For small circuits, state-vector simulator plugins may be more suitable. The supported method is Matrix Product State (MPS). This device does not currently support finite shots. At present, the supported measurement types are expectation values and the supported differentiation methods are ``parameter-shift`` or ``finite-diff``. Note that only 1,2-wires gates and gates that can be decomposed by PennyLane into 1,2-wires gates are supported.
```

> * ``lightning.tensor``: is a tensor network simulator based on the `NVIDIA cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_ (requires NVDIA GPUs with SM 7.0 or greater). This device is designed to simulate large-scale quantum circuits using tensor networks. For small circuits, state-vector simulator plugins may be more suitable. The supported method is Matrix Product State (MPS). This device does not currently support finite shots. At present, the supported measurement types are expectation values and the supported differentiation methods are ``parameter-shift`` or ``finite-diff``. Note that only 1,2-wires gates and gates that can be decomposed by PennyLane into 1,2-wires gates are supported.

Reviewer 2: ```suggestion
* ``lightning.tensor``: is a tensor network simulator based on the `NVIDIA cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_ (requires NVIDIA GPUs with SM 7.0 or greater). This device is designed to simulate large-scale quantum circuits using tensor networks. For small circuits, state-vector simulator plugins may be more suitable. The supported method is Matrix Product State (MPS). This device does not currently support finite shots. Currently, the supported measurement types are expectation values and the supported differentiation methods are ``parameter-shift`` and ``finite-diff``. Note that only 1,2-wire gates and gates that can be decomposed by PennyLane into 1,2-wire gates are supported.
```

.. header-end-inclusion-marker-do-not-remove

The following table summarizes the supported platforms and the primary installation mode:

+-----------+---------+--------+-------------+----------------+-----------------+----------------+----------------+
|           | L-Qubit | L-GPU  | L-GPU (MPI) | L-Kokkos (OMP) | L-Kokkos (CUDA) | L-Kokkos (HIP) |    L-Tensor    |
+===========+=========+========+=============+================+=================+================+================+

> | Linux x86 | pip     | pip    | source      | pip            | source          | source         |     source     | 

Reviewer: ```suggestion
| Linux x86 | pip     | pip    | source      | pip            | source          | source         |     source     |
```

> | Linux x86 | pip     | pip    | source      | pip            | source          | source         |     source     |

Reviewer 3: Won't we be shipping wheels?
```suggestion
| Linux x86 | pip     | pip    | source      | pip            | source          | source         |     pip     |
```

Author: Not a goal of Q2 but a goal of Q3

+-----------+---------+--------+-------------+----------------+-----------------+----------------+----------------+

> | Linux ARM | pip     | source |             | pip            | source          | source         |                |

Reviewer 3: I guess we have not tested it, but could we quickly check whether it works on our GraceHopper machine?
```suggestion
| Linux ARM | pip     | source |             | pip            | source          | source         |  source   |
```

Author: There are randomly failures for both LGPU and LTensor when @maliasadi tested them on GH. It's on the roadmap of Q3.

+-----------+---------+--------+-------------+----------------+-----------------+----------------+----------------+
| Linux PPC | pip     | source |             | pip            | source          | source         |                |
+-----------+---------+--------+-------------+----------------+-----------------+----------------+----------------+
| MacOS x86 | pip     |        |             | pip            |                 |                |                |

> +-----------+---------+--------+-------------+----------------+-----------------+----------------+----------------+   

Reviewer: ```suggestion
+-----------+---------+--------+-------------+----------------+-----------------+----------------+----------------+
```

.. installation_LKokkos-end-inclusion-marker-do-not-remove

.. installation_LTensor-start-inclusion-marker-do-not-remove

Lightning-Tensor installation
*****************************
Lightning-Tensor requires CUDA 12 and the `cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_ (only the `cutensornet <https://docs.nvidia.com/cuda/cuquantum/latest/cutensornet/index.html>`_ library is required).
The SDK may be installed within the Python environment ``site-packages`` directory using ``pip`` or ``conda`` or the SDK library path appended to the ``LD_LIBRARY_PATH`` environment variable.
Please see the `cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_ install guide for more information.

Install Lightning-Tensor from source
====================================
Lightning-Qubit should be installed before Lightning-Tensor:

.. code-block:: console

    git clone https://github.com/PennyLaneAI/pennylane-lightning.git
    cd pennylane-lightning
    pip install -r requirements.txt

>     PL_BACKEND="lightning_qubit" pip install .

Reviewer 2: It might be useful for user to know that they may install `lightning.qubit` without binaries.
It is much quicker this way. :slightly_smiling_face: 
```suggestion
    SKIP_COMPILATION=True  PL_BACKEND="lightning_qubit" pip install .
```

.. installation_LKokkos-end-inclusion-marker-do-not-remove

.. installation_LTensor-start-inclusion-marker-do-not-remove

Lightning-Tensor installation
*****************************
Lightning-Tensor requires CUDA 12 and the `cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_ (only the `cutensornet <https://docs.nvidia.com/cuda/cuquantum/latest/cutensornet/index.html>`_ library is required).

> The SDK may be installed within the Python environment ``site-packages`` directory using ``pip`` or ``conda`` or the SDK library path appended to the ``LD_LIBRARY_PATH`` environment variable.

Reviewer 4: I am not sure this line is more helping than confusing, should we just leave it with the link to the SDK installation guide below? (I actually dont know what is best and leave it up to you)

> Please see the `cuQuantum SDK`_ install guide for more information.

Reviewer: ```suggestion
Please see the `cuQuantum SDK <https://developer.nvidia.com/cuquantum-sdk>`_ install guide for more information.
```

Install Lightning-Tensor from source
====================================

> To install Lightning-Tensor from the package sources using the direct SDK path, Lightning-Qubit should be install before Lightning-Tensor:

Reviewer 4: ```suggestion
To install Lightning-Tensor from the package sources using the direct SDK path, Lightning-Qubit should be installed before Lightning-Tensor:
```

> To install Lightning-Tensor from the package sources using the direct SDK path, Lightning-Qubit should be installed before Lightning-Tensor:

Reviewer 2: Could you please rephrase that? This sentence is weird.

.. code-block:: console

    git clone https://github.com/PennyLaneAI/pennylane-lightning.git
    cd pennylane-lightning

>     pip install -r requirements.txt

Reviewer 3: `requirements-dev.txt` includes `cutensornet-cu12`
```suggestion
    pip install -r requirements-dev.txt
```

Author: yea, `cutensornet-cu12` is included in `requirements-dev.txt`.  But the installation of `lightning_qubit` doesn't require `cutensornet-cu12` or `custatevec-cu12` listed in `requirements-dev.txt`.

>     pip install -r requirements.txt

Reviewer 3: ```suggestion
    python -m venv venv && source venv/bin/activate && pip install -r requirements.txt
```

Author: Let's ask users to do everything in a virtual env explicitly.

>     PL_BACKEND="lightning_qubit" pip install -e .

Reviewer 3: I wonder whether we should remove `-e` flags throughout since this is instructing users as well as developers.  
```suggestion
    PL_BACKEND="lightning_qubit" pip install .
```

Then the `cutensornet`_ library can be installed and set a ``CUQUANTUM_SDK`` environment variable.

.. code-block:: console

    pip install cutensornet-cu12

>     export CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/lib')")

Reviewer 3: I don't think that is necessary, but doesn't hurt I guess.
```suggestion
```
